src/transformers/pipelines/question_answering.py

```diff
@@ -509,8 +509,8 @@ def preprocess(self, example, padding="do_not_pad", doc_stride=None, max_questio
     def _forward(self, inputs):
         example = inputs["example"]
         model_inputs = {k: inputs[k] for k in self.tokenizer.model_input_names}
-        start, end = self.model(**model_inputs)[:2]
```

Comment:
Model outputs can either be regular tuples or our hybrid subclass of dicts that also support slice indexing. So here there should be a test on `outputs = self.model(**model_inputs)`. If it satisfies `isinstance(outputs, tuple)` we should leave the code as is, and in the else branch we can use your changes.

Reply:
thanks for the comments. fixed